### PR TITLE
VACMS-19512: Drupal UX for phone numbers

### PR DESCRIPTION
**Pull request by davidmpickett**
## Description

Relates to #19512 
Also fixing #16150 as a bonus

## Testing done
Manual in Tugboat

## Screenshots

Entity & state | Current UI | Updated UI
-- | -- | --
Staff profile (no phone) |Screenshot 2024-10-25 at 12 53 13 PM |Screenshot 2024-10-24 at 5 16 25 PM
Staff profile (with phone) |Screenshot 2024-10-24 at 5 19 03 PM |Screenshot 2024-10-24 at 5 31 56 PM
VAMC Billing & insurance (with phone) |Screenshot 2024-10-25 at 12 15 41 PM |Screenshot 2024-10-24 at 5 52 26 PM
VAMC Billing & insurance (no phone) |Screenshot 2024-10-25 at 12 15 29 PM |379959166-f3f7b956-3fa8-448e-b145-58b646920e8f
VAMC Facility mental health (with phone) |Screenshot 2024-10-25 at 12 11 38 PM |Screenshot 2024-10-25 at 12 49 46 PM
VAMC Facility mental health (no phone) |Screenshot 2024-10-25 at 12 11 19 PM |Screenshot 2024-10-25 at 12 49 32 PM
VAMC Facility mental health (FEATURE TOGGLE OFF) |Screenshot 2024-10-25 at 2 24 58 PM |Screenshot 2024-10-25 at 2 25 52 PM

## QA steps

What needs to be checked to prove this works?
What needs to be checked to prove it didn't break any related things?
What variations of circumstances (users, actions, values) need to be checked?

### Once per day setup
1. [Log in](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/) as an admin
2. Go to [Feature Toggle page](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/admin/config/system/feature_toggle) 
   - [x] Check FEATURE_TELEPHONE_MIGRATION_V1
   - [x] Click Save
3. Run Telephone migration (optional)
   - [x] Go to [Dashboard](https://tugboat.vfs.va.gov/671ac5aa25a4ad7d76bce8f3)
   - [x] Run each command below separately in php terminal
   - [x] Total migration time should be 30-40 minutes.

```
time drush codit-batch-operations:run MigratePhoneFieldToParagraph
time drush codit-batch-operations:run MigrateVamcBillingInsurancePhoneFieldToParagraph
time drush codit-batch-operations:run MigrateVamcFacilityMentalHealthPhoneFieldToParagraph
```

### Check Staff Profile
1. [Log in to tugboat](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/) as a VAMC editor or Content Admin
2. Go to Content > Add Content > [Staff Profile](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/node/add/person_profile)
3. Save a staff profile with no phone number
   - [x] Fill in the required fields (First name, last name, section, related office)
   - [x] In the contact information box, notice that by default, no phone number has been added
   - [x] Add a revision message and save as draft
4. Save a staff profile with a phone number
   - [x] Edit the staff profile you just created
   - [x] In the contact information box click the Add Phone number button
   - [x] Before filling out any of the phone number sub-fields, add a revision message and attempt save as draft
   - [x] See validation error for not filling out the required phone number sub-field 
   - [x] Fill in the required fields
   - [x] Save as draft
5. Remove a phone number from a staff profile
   - [x] Edit the staff profile again
   - [x] In the contact information box click the Remove button
   - [x] Click confirm removal button (this is technically optional)
   - [x] Add a revision message and save as draft

### Check VAMC System Billing & Insurance
1. [Log in to tugboat](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/) as a Content Admin
2. Go to Content > Add Content > [VAMC System Billing and Insurance](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/node/add/vamc_system_billing_insurance)
3. Save a Billing & Insurance page with a phone number
   - [x] Notice that in the "For inquiries by phone about copay balance" section, a phone number is added by default
   - [x] Before filling out any of the phone number sub-fields, add a revision message and attempt save as draft
   - [x] See validation error for not filling out the required phone number sub-field 
   - [x] Fill in other required fields
   - [x] Save as draft
4. Remove a phone number from a Billing & Insurance page
   - [x] Edit the page you just created
   - [x] In the "For inquiries by phone about copay balance" section click the Remove button
   - [x] Click confirm removal button (this is technically optional)
   - [x] Add a revision message and save as draft
5. Add a phone number 
   - [x] Edit the page you just created
   - [x] In the "For inquiries by phone about copay balance" section click the Add Phone number button
   - [x] Confirm that the correct phone number fields populate
6. Verify change for [16150](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/16150) 
    - [x] Edit the page you just created
    - [x] Notice that there is no longer an empty "Cashier's office" accordion

### Check VAMC Facility - Mental Health Phone Number
1. [Log in to tugboat](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/) as a Content Admin
2. Go to Content > Add Content > [VAMC Facility](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/node/add/health_care_local_facility)
3. Save a VAMC facility page with a phone number
   - [x] Notice that in the "Mental health phone number" section, a phone number is added by default
   - [x] Before filling out any of the phone number sub-fields, add a revision message and attempt save as draft
   - [x] See validation error for not filling out the required phone number sub-field 
   - [x] Fill in other required fields
   - [x] Save as draft
4. Remove a phone number from a VAMC facility page
   - [x] Edit the page you just created
   - [x] In the "Mental health phone number" section click the Remove button
   - [x] Click confirm removal button (this is technically optional)
   - [x] Add a revision message and save as draft
5. Add a phone number 
   - [x] Edit the page you just created
   - [x] In the "Mental health phone number" section click the Add Phone number button
   - [x] Confirm that the correct phone number fields populate

### Check  VAMC Facility - Mental Health Phone Number TOGGLE OFF
1. [Log in](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/) as an admin
2. Go to [Feature Toggle page](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/admin/config/system/feature_toggle) 
   - [ ] Uncheck FEATURE_TELEPHONE_MIGRATION_V1
   - [ ] Click Save
3. Go to Content > Add Content > [VAMC Facility](https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/node/add/health_care_local_facility)
4. Notice the slightly changed interface compared to prod

### Check other phone number paragraph types to make sure we didn't break their styling
- VAMC System VA Police
- VBA Facility Service (service location)

### Definition of Done

- [x] Automated tests have passed.
- [x] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### CODE REVIEW

**Comment by jilladams** [comment]
# QA notes

## Basic behavior
- Content with no phone number: add phone, save:  🟢
- Content with a phone number: remove phone number, save: 🟢 

## Label field weirdness 🔴 
On Staff profile, I removed an existing phone number, added a new one and saved with max char length to test. After removing phone, clicking to Add phone, I see a Label field (across all content types). To reproduce: 

* Remove an existing phone # 
* Add phone - new fields appear: Type, Phone Number, Extension number, Label. 
  * Refresh page without populating fields: 
    * Label field disappears
  * OR, save content in the field. 
    * Node view shows the Label field value
    * Node edit hides the Label field

https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/node/73936
Screenshot 2024-10-25 at 3 31 43 PM

https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/node/73936/edit
Screenshot 2024-10-25 at 3 31 37 PM

Test content (FE is building on Fri pm and can be verified on Monday):
* Staff profile node: https://web-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/staff-profiles/esther-reece
  * Staff profile FE: https://web-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/staff-profiles/esther-reece
* B&I node: https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/central-western-massachusetts-health-care/billing-and-insurance
  * B&I FE: https://web-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/central-western-massachusetts-health-care/billing-and-insurance
* VA clinic MH: https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/spokane-health-care/locations/spokane-valley-va-clinic
  * FE: https://web-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/spokane-health-care/locations/spokane-valley-va-clinic
* Mobile clinic: https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/southern-arizona-health-care/locations/tucson-va-mobile-clinic
  * https://web-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/southern-arizona-health-care/locations/tucson-va-mobile-clinic
* VAMC: https://pr19607-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/alaska-health-care/locations/joint-base-elmendorf-richardson-va-medical-center
  * https://web-h5ubpjxulma3zyqeq8hayikh80scnqhu.ci.cms.va.gov/alaska-health-care/locations/joint-base-elmendorf-richardson-va-medical-center

This probably blocks the PR, until we prove whether the label prints in the FE or not.

## Remove confirmation / truncated text 🟡 
Removal confirmation is technically optional per the QA steps. 

When I use the "remove" button to remove, I get a confirmation prompt with truncated text. Loom that shows the experience: https://www.loom.com/share/7f203fbc6650489288d51f85a0e3dd09

Screenshot 2024-10-25 at 3 15 32 PM

This doesn't have to block the PR, but is not great UX debt to add while trying to improve the experience, so I do think it needs follow up. If the confirmation is optional, can we remove it? Or do we need to keep it and address the truncated text? (Updating `.js .paragraph-type-top .paragraph-type-title` to > 25% displays the full text.) 

## Empty Cashier's office accordion on B&I
Observed that Cashier's office accordion is empty. That's a documented defect: #16150

**Comment by jilladams** [comment]
I'm a goofy person, and we cannot test what happens in the fe, because the FE isn't using the new phone number structure yet. I think. Not sure what that means for troubleshooting / fixing labels and whether or not this can merge.

**Comment by davidmpickett** [comment]
> ## Label field weirdness 🔴
> On Staff profile, I removed an existing phone number, added a new one and saved with max char length to test. After removing phone, clicking to Add phone, I see a Label field (across all content types). To reproduce:
> 
> * Remove an existing phone #
> * Add phone - new fields appear: Type, Phone Number, Extension number, Label.
>   
>   * Refresh page without populating fields:
>     
>     * Label field disappears
>   * OR, save content in the field.
>     
>     * Node view shows the Label field value
>     * Node edit hides the Label field
> 

I also noticed this behavior when I was doing QA. However, this behavior is not specific to my PR. I flipped the feature toggle on [my demo tugboat](https://cms-sivbehcvqkui7ce9b67dwanfbohmenin.demo.cms.va.gov/) and confirmed that the label field shows up if you add, remove, and re-add.

Screenshot 2024-10-28 at 10 02 58 AM

Fixing this should be a separate ticket.

**Comment by davidmpickett** [comment]
> ## Remove confirmation / truncated text 🟡
> Removal confirmation is technically optional per the QA steps.
> 
> When I use the "remove" button to remove, I get a confirmation prompt with truncated text. Loom that shows the experience: https://www.loom.com/share/7f203fbc6650489288d51f85a0e3dd09
> 
> Screenshot 2024-10-25 at 3 15 32 PM
> 
> This doesn't have to block the PR, but is not great UX debt to add while trying to improve the experience, so I do think it needs follow up. If the confirmation is optional, can we remove it? Or do we need to keep it and address the truncated text? (Updating `.js .paragraph-type-top .paragraph-type-title` to > 25% displays the full text.)
> 

I just committed a change to increase that to 40%, but then that makes this PR need CMS platform review since that's changing a more fundamental property in the system. I'm also not sure how we would go about verifying that increasing this percentage doesn't create problems elsewhere in the CMS.

**Comment by jilladams** [comment]
I think this is now ready for code review, @omahane

**Comment by davidmpickett** [comment]
> ## Empty Cashier's office accordion on B&I
> Observed that Cashier's office accordion is empty. That's a documented defect: #16150

This change was pretty easy to pull in to my PR. 
- I removed the empty accordion
- I updated the top of this PR to reflect it also resolves 16150
- I updated the QA steps to reflect noticing that fix

File: config/sync/core.entity_form_display.node.person_profile.default.yml [hunk 1]
```diff
@@ -258,22 +258,16 @@ content:
       placeholder: ''
     third_party_settings: {  }
   field_telephone:
-    type: paragraphs
+    type: entity_reference_paragraphs
```
Comment (jv-agile6) [inline config/sync/core.entity_form_display.node.person_profile.default.yml:261]: Added entity reference paragraph

File: config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml [hunk 2]
```diff
@@ -248,22 +248,16 @@ content:
       placeholder: ''
     third_party_settings: {  }
   field_telephone:
-    type: paragraphs
+    type: entity_reference_paragraphs
```
Comment (jv-agile6) [inline config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml:237]: added entity reference paragraph

**Review by jv-agile6 — COMMENTED** [review]
Reviewed code configuration in core.entity_form_display.node.person_profile.default.yml and config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml for form and logic. Followed additions and subtractions to code and found no defects.

**Review by jv-agile6 — DISMISSED** [review]
Code review completed. Approval granted.

**Comment by laflannery** [comment]
1. Before a phone number is added, `Phone number` is correctly styled bold. When I add a phone number, this is no longer bold, which makes it harder to pick out when I want to go find it again and edit it:

**Before adding**
image

**After adding**
image

2. Italics are not in the DS, this should be updated to plain text:
image

3. The `Remove` button should be updated to say `Remove Phone number` or have an `aria-describedby="edit-field-telephone-0-top-paragraph-type-title--0PvCeESrLmY"` attribute for AT users at least. Currently, because it's be disassociated from the table, I have no idea what I'm removing so I need a new association.

4. The whole thing is not wrapped in fieldset/legend markup, which is actually because it's typically contained within a larger fieldset, except on the Billing and Insurance content type. Are we able to change this so that all the phone fields are wrapped in a `<fieldset> tag and "Phone number" is the `<legend>`? Simplified example below:
image

**Comment by omahane** [comment]
@laflannery I can't seem to figure out where your screenshot is from that shows the phone number without a fieldset.

# Fieldsets
## Staff profile
Screenshot 2024-12-04 at 5 27 01 PM

## VAMC Facility
Screenshot 2024-12-04 at 5 27 08 PM

## VAMC System Billing and Insurance
Screenshot 2024-12-04 at 5 27 12 PM

All of these have a fieldset. I wonder if the Staff Profile one (which is what I think your screenshot is from) was before Dave added the fieldset.

# Remove button
How important is it to change this remove button to include what is being removed? We would need to either custom code this or try to find/implement an upstream patch on a widget that is legacy, as the name implies and is thus not in development any longer.

**Comment by omahane** [comment]
@laflannery I have made "Phone number" when there's a phone and have made the italics normal when there's no phone. Please confirm when you have a minute when this is rebuilt.

**Comment by laflannery** [comment]
Sorry, one more thing. On VAMC Facility content type. When the paragraph field is present I see "Phone number" but then when I remove it this label changes to "Mental health phone number". Was that intentional? Can it remain "Phone number"?
Screenshot 2024-12-05 at 3 15 27 PM

**Comment by omahane** [comment]
> Sorry, one more thing. On VAMC Facility content type. When the paragraph field is present I see "Phone number" but then when I remove it this label changes to "Mental health phone number". Was that intentional? Can it remain "Phone number"? !

This is a known bug in the old paragraph widget, which we are using here:
https://www.drupal.org/project/paragraphs/issues/3067178

If we want, we could go back to using the newer one, which has its own UX oddities.

**Comment by laflannery** [comment]
@omahane Then that's fine, it's not a deal breaker, thanks!

File: config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml [hunk 3]
```diff
@@ -121,20 +120,6 @@ third_party_settings:
         description: ''
         required_fields: false
         formatter: closed
-    group_cashier_s_office:
```
Comment (dsasser) [inline config/sync/core.entity_form_display.node.vamc_system_billing_insurance.default.yml:124]: Noting this change is expected as it resolves #16150 which is included in this PR.

**Review by dsasser — APPROVED** [review]
Code changes look good!

**Comment by omahane** [comment]
I can't save a new VAMC System Billing and Insurance:
Screenshot 2024-12-10 at 12 30 40 PM

It's fine. The phone validation worked correctly.

**Comment by omahane** [comment]
I didn't see the point in testing with the toggle off, as we've already migrated and have moved over to the new paragraph phone field in production.

**Pull request merged**